### PR TITLE
Add place for community solutions and Q 2

### PR DESCRIPTION
**Pull request by dalurness**
- Added Q2
- Added pages for the community to put their coded solutions. It displays their username with a link to their profile, as well as displaying their solution. All they need to do is add a markdown file named with their username for it to automatically be included. Obviously it would have been nice to be able to nest them inside of the existing question directories in the content directory, but that is not currently supported by Astro. There is an Issue for it already though and something they are looking to support in the future.

The solution page has the question below the solution for reference.

image
image

### CODE REVIEW

File: src/layouts/Navbar.astro
Comment (janka102) [inline src/layouts/Navbar.astro:1]: Perhaps this should be a component instead since it doesn't have a slot

**Comment by janka102** [comment]
Do we want a way for the same user to be able to submit multiple solutions? I think that could be good, since we want to reuse the same questions for multiple years. Maybe we could allow the user to name their solutions, then they could even have multiple solutions for one year if they want.
We could even group solutions by year. One way would be add a date field to each solution (or maybe just year without month and day)

A future idea I had, just writing it down so I don't forget:
- to make it super simple, have a script that automatically adds a given file as a solution

File: src/layouts/Navbar.astro
Comment (dalurness) [reply src/layouts/Navbar.astro:1]: switched it to a component

**Comment by dalurness** [comment]
updated to allow multiple solutions per person

**Comment by janka102** [comment]
As discussed, you can put multiple code blocks in the one file for multiple solutions, you can even put whatever else you want in there like headings and paragraphs to label and describe your solution or auxiliary information.

**Review by janka102 — APPROVED** [review]

**Pull request merged**